Storage/Storage/CoreData/CoreDataManager.swift

```diff
@@ -59,71 +62,76 @@ public final class CoreDataManager: StorageManagerType {
     /// Returns the Storage associated with the View Thread.
     ///
     public var viewStorage: StorageType {
-        return persistentContainer.viewContext
+        return syncQueue.sync {
+            return persistentContainer.viewContext
+        }
     }
 
     /// Returns a shared derived storage instance dedicated for write operations.
     ///
     public lazy var writerDerivedStorage: StorageType = {
-        let childManagedObjectContext = NSManagedObjectContext(concurrencyType: .privateQueueConcurrencyType)
-        childManagedObjectContext.parent = persistentContainer.viewContext
-        childManagedObjectContext.mergePolicy = NSMergeByPropertyObjectTrumpMergePolicy
-        return childManagedObjectContext
+        return syncQueue.sync {
```

Comment:
Instead of using `DispatchQueue` and closures, how about making `CoreDataManager` an `actor`?

Reply:
This is another possibility, but do you see a real benefit over `DispatchQueue`? I'm concerned that making this class an actor might entail many changes elsewhere.

Reply:
A benefit is to have a more modern syntax with no requirement for using closures, making the code more readable. I understand the potential side effects of changes required at call sites though.

Reply:
I believe, and let me know if you agree, that if this PR is merged and it fixes the crashes on Core Data, it's worth investing some time later in migrating it to `actor`, unless the impact is huge.